### PR TITLE
Add Remove Brackets Command

### LINKED ISSUE #10: Add ability to parse quotes and brackets in comments
Standalone issue created based on feedback from #8

### LINKED ISSUE #49: (Option to) delete highlighted brackets rather than "closest"
Hi there, thanks for this extension!  I would find it convenient if the shortcut deleted highlighted brackets rather than directly enclosing, so that I have direct visual feedback about what is to be removed with the next keystroke

### LINKED ISSUE #55: [Bug] When the string contains `@`, Bracketeer warns "No quotes to modify found."
Repro case:

```json
{
  "@foo": "foo"
}
```

Please use command `bracketeer.selectQuotesContent` to select `@foo`.

Quite often npm package names can start with `@`, e.g.

```json
{
  "@vue/cli-plugin-babel": "~4.5.13",
  "@vue/cli-plugin-eslint": "~4.5.13",
  "@vue/cli-plugin-router": "~4.5.13",
  "@vue/cli-plugin-vuex": "~4.5.13",
  "@vue/cli-service": "~4.5.13",
  "@vue/compiler-sfc": "^3.1.5",
  "@vue/eslint-config-prettier": "^6.0.0"
}
```

So it's currently impossible to select them with Bracketeer.

### PR DESCRIPTION
**Pull request by zardoy**
Description: deletes **highlighted** brackets pair in current editor.

fixes https://github.com/Pustelto/Bracketeer/issues/49
fixes https://github.com/Pustelto/Bracketeer/issues/10
fixes https://github.com/Pustelto/Bracketeer/issues/55

### Motivation

Bracketeer has hard limitation as extensions can't read language configurations in the web (in case of Bracketeer ~~all~~ some brackets are hardcoded e.g. vscode dynamic `<` and `>` bracket matching).

As a benefit of using native bracket model, there is small, but noticeable performance improvement.
I compared it on the contents of this file: https://github.com/microsoft/TypeScript/blob/main/lib/tsserverlibrary.d.ts

Bracketeer: latency 40ms-1s ~150ms for first time in position, then ~40ms
New command: always 2-11ms

### CODE REVIEW

**Comment by zardoy** [comment]
//c @hediet

**Comment by hediet** [comment]
Related: https://github.com/microsoft/vscode/pull/134371

It would be cool if ctrl+delete would delete the bracket before the cursor and it's matching counterpart.

**Comment by zardoy** [comment]
@hediet is this ready to merge, or there is anything else to be done?

I also wonder shouldn't this command be renamed to *Delete Brackets* instead?

File: src/vs/editor/contrib/bracketMatching/browser/bracketMatching.ts [hunk 1]
```diff
@@ -251,6 +270,32 @@ export class BracketMatchingController extends Disposable implements IEditorCont
 			this._editor.revealRange(newSelections[0]);
 		}
 	}
+	public replaceBrackets(openingContent: string, closingContent: string, editSource?: string): void {
```
Comment (hediet) [inline src/vs/editor/contrib/bracketMatching/browser/bracketMatching.ts:273]: Please remove the first two parameters, they always seem to be `''`.

**Review by hediet — APPROVED** [review]

File: src/vs/editor/contrib/bracketMatching/browser/bracketMatching.ts (hunk 1 above)
Comment (zardoy) [reply src/vs/editor/contrib/bracketMatching/browser/bracketMatching.ts]: Thanks for review! Okay, I will revert it to its original implementation.

I changed implementation in https://github.com/microsoft/vscode/pull/160270/commits/b2bc6c55e121389824b12a46545c9ed7380d66cc as I thought it'd better for infrustructure to have general method instead, so command to replace brackets would be implemented even more cleanly... Sorry, I'll try to not include unnecessary changes until needed next time!

Btw, how this editSource in `executeEdits` is used? Can I see it in some output channel when doing text changes?

**Review by ulugbekna — APPROVED** [review]

**Pull request merged**